lib/rubocop/cop/lint/literal_in_interpolation.rb

```diff
@@ -29,7 +28,9 @@ def on_dstr(node)
         private
 
         def special_string?(node)
-          node.type == :str && !node.loc.respond_to?(:begin)
+          # handle strings like __FILE__
+          (node.type == :str && !node.loc.respond_to?(:begin)) ||
+            node.loc.expression.is?('__LINE__')
```

Comment:
`__method__` is just a `Kernel` method. 

Not sure what the technical term for something like `__FILE__` and `__LINE__` is. If there's something common we can use it, otherwise `special_string?` and `special_int?` will do. I came up with `special_string?` just because I didn't know how exactly to name this...